docs/DeveloperGuide.md

```diff
@@ -1,34 +1,315 @@
+creation:
+
+1) `Module Code` <br> Example: `CS2113T`
+2) `Task Description` <br> Example: `Finish v2.0`
+3) `Date` <br> Example: `2021-03-21`
+4) `Time` <br> Example: `23:59`
+5) `Message` <br> Example: `Good job!`
+
+Given below is the sequence diagram for the feature:  
+![Diagram](diagrams/addNewTask.png)
+
+A general explanation of how this feature works:
+
+When the user calls this feature, the application will prompt them to input the arguments required
+for task creation. This new task will then be stored into an `ArrayList` for future reference. The
+user has to input a valid format for the __date__ and __time__ field (refer to section 3.3.1 of User
+Guide or the example above), and our application will convert it to a more readable format for the
+user. Using the example above,
+
+- `2021-03-21` will be outputted as `Mar 21 2021`
+- `23:59` will be outputted as `11:59 PM`
+
+The user input for `Message` (argument 5 above) will be printed out when the user deleted the task,
+signalling completion.
+
+### 4.04 Delete a Task
+
+This feature allows the user to delete a task of type `task`, `assignment`, `midterm`, `final exam`
+from a task list.
+
+A general explanation of how this feature works:
+
+When the user calls this feature, the application will prompt them to choose the __task
+type__ (`task`, `assignment`, `midterm`, `final exam`) of the task they want to delete. All existing
+tasks in the `ArrayList` for the task type they chose will then be printed out, and the user has to
+input the index of the task they want to delete. The task they select will then be deleted from
+the `ArrayList`.
+
+The `Message` that the user inputted upon creation of this particular task will then be printed out.
+
+### 4.05 Add Zoom Link
+
+Given below is the sequence diagram for the feature:  
+![Diagram](diagrams/addZoomLink.png)
+
+A general 4-step explanation of how this feature works is shown below:
+
+Step 1. The ZoomLinkInfo#addZoomLink() method calls AddTask#printAndGetModule() which asks the user
+to input the number corresponding to the module code they want, and returns the module code.
+
+Step 2. The ZoomLinkInfo#addZoomLink() method calls other methods in the Ui class to determine the
+link, and whether a password is required for the Zoom meeting.
+
+*Note: If the user does not require the zoom link to be tagged to a module,
+AddTask#printAndGetModule() in Step 1 would return an empty string, and the module code would be
+initialised to "Zoom link has no module code".
+
+Step 3. The ZoomLinkInfo#addZoomLink() method then calls ModuleInfo#getModule() in order to retrieve
+the module object which is stored in the ModuleInfo class.
+
+Step 4. Module#setZoomLink then sets the `zoomLink` attribute of the module object to the zoom link
+that was entered.
+
+### 4.06 Add Module Components
+
+This feature allows user to add user-defined module components and its related weightage for an
+existing module.
+
+![Diagram](diagrams/addComponent.png)  
```

Comment:
In the sequence diagram of addComponent, it would be good if it is shown where "command" is used for after it was returned to Component. (or at least  show that it is subsequently returned to ModuleInfo & Duke if "command" is used for other parts) :)